frontend_app_gallery/src/container/Gallery.jsx

```diff
@@ -154,20 +164,34 @@ class Gallery extends React.Component {
     // app crash telling it cannot render a Link outside a router
     // see https://github.com/tracim/tracim/issues/1637
     // GLOBAL_dispatchEvent({type: 'setBreadcrumbs', data: {breadcrumbs: breadcrumbsList}})
-    this.setState({ breadcrumbsList: breadcrumbsList })
+    return breadcrumbsList
   }
 
   loadFolderDetail = async (workspaceId, folderId) => {
```

Comment:
you have a function loadFolderDetail and getFolderDetail, these 2 names are too ambiguous. We need better names.
From my understanding, this function will load the parents id up to the root workspace. It's name should show that.